backend/src/db/daos/noteDao.js

```diff
@@ -8,7 +8,17 @@ import Group from "../models/group";
  * @param {String} role role of the note
  * @returns
  */
-const createNote = async (groupId, title, role, text = "") => {
+const createNote = async (groupId, title, email, text = "") => {
+  const dbGroup = await Group.findById(groupId);
```

Comment:
nit: the auth check could be abstracted out to comply with DRY.

E.g. method that retrieves the requesting user's role could be a potential path.

Reply:
I've made and am using a function for this already, you could just move it to the groupDao file and then use it from there.

https://github.com/UoaWDCC/VPS/blob/58baaf46e9eb3aeacfc9e7b7ecd5316cfc8ec44b/backend/src/routes/api/navigate/group.js#L38

Reply:
Cool, I will try to write a method to check if the user is within a group because it's been used alot